caravel/assets/javascripts/explore/explore.jsx

```diff
@@ -345,23 +356,40 @@ function initComponents() {
     />,
     queryAndSaveBtnsEl
   );
-
-  const exploreActionsEl = document.getElementById('js-explore-actions');
-  ReactDOM.render(
-    <ExploreActionButtons
-      canDownload={exploreActionsEl.getAttribute('data-can-download')}
-      slice={slice}
-    />,
-    exploreActionsEl
-  );
+  renderExploreActions();
 }
 
+let exploreController = {
+  type: 'slice',
+  done: (sliceObj) => {
+    slice = sliceObj;
+    renderExploreActions();
+    const cachedSelector = $('#is_cached');
+    if (slice.data !== undefined && slice.data.is_cached) {
+      cachedSelector
+      .attr(
+        'title',
+        `Served from data cached at ${slice.data.cached_dttm}. Click [Query] to force refresh`)
+      .show()
+      .tooltip('fixTitle');
+    } else {
+      cachedSelector.hide();
+    }
+  },
+  error: (sliceObj) => {
+    slice = sliceObj;
+    renderExploreActions();
+  },
+};
+exploreController = Object.assign({}, utils.controllerInterface, exploreController);
+
+
 $(document).ready(function () {
   const data = $('.slice').data('slice');
 
   initExploreView();
 
-  slice = px.Slice(data);
+  slice = px.Slice(data, exploreController);
 
   $('.slice').data('slice', slice);
```

Comment:
good catch, I'm removing it and making sure things still work. We're re-writing the explore view to react so I don't want to go too far in the refactoring, but that's an easy win.